src/Raven.Server/RavenServer.cs

```diff
@@ -929,45 +935,188 @@ internal AuthenticateConnection AuthenticateConnectionCertificate(X509Certificat
-                    }
-                    else
+                    try
                     {
-                        var definition = JsonDeserializationServer.CertificateDefinition(cert);
-                        authenticationStatus.Definition = definition;
-                        if (definition.SecurityClearance == SecurityClearance.ClusterAdmin)
-                        {
-                            authenticationStatus.Status = AuthenticationStatus.ClusterAdmin;
-                        }
-                        else if (definition.SecurityClearance == SecurityClearance.ClusterNode)
-                        {
-                            authenticationStatus.Status = AuthenticationStatus.ClusterAdmin;
-                        }
-                        else if (definition.SecurityClearance == SecurityClearance.Operator)
+                        var certKey = Constants.Certificates.Prefix + certificate.Thumbprint;
+                        var cert = ServerStore.Cluster.GetCertificateByPrimaryKey(ctx, certKey) ??
+                                   ServerStore.Cluster.GetLocalState(ctx, certKey);
+
+                        if (cert == null)
                         {
-                            authenticationStatus.Status = AuthenticationStatus.Operator;
+                            // The certificate is not explicitly registered in our server, let's see if we have a certificate
+                            // with the same public key pinning hash.
+                            var pinningHash = CertificateUtils.GetPublicKeyPinningHash(certificate);
+                            var certsWithSameHash = ServerStore.Cluster.GetCertificatesByPinningHashSortedByExpiration(ctx, pinningHash);
+
+                            if (certsWithSameHash.Count > 0)
+                            {
+                                // Hash is good, let's validate it was signed by same issuer, otherwise users can use the private key to register a new cert with a different issuer.
+                                if (CertHasKnownIssuer(certificate, certsWithSameHash) == false)
+                                {
+                                    if (Logger.IsOperationsEnabled)
+                                        Logger.Operations($"You tried to use a certificate: '{certificate.Subject} ({certificate.Thumbprint})' which is not " +
+                                                          "registered in the cluster explicitly but is trusted implicitly by its HTTP Public Key Pinning Hash (HPKP). " +
+                                                          "Your certificate was signed by an unknown issuer. This is not allowed - closing the connection. " +
+                                                          "To fix this, the admin can register the thumbprint of the *issuer* certificate (e.g. Let's Encrypt) in the 'Security.WellKnownIssuers.Admin' configuration entry.");
```

Comment:
Provide the value that needs to be set here.

Reply:
The message needs to also be shown to the user

Reply:
done